### PR TITLE
[READY] Venus human trap fix 3.0: Allow ghosts to possess them even when obscured, use better spawning system

### PR DESCRIPTION
**Pull request by RandomGamer123**
## About The Pull Request
Moves venus human traps to use the /obj/effect/mob_spawn/ghost_role/ system of spawning similar to things like spiders. Effects of this include: Having a notification to ghosts when one is ready to be spawned, having venus human traps respect role bans, and allowing them to be possessed from the ghost role menu.
Also adds poll ignore and role defines for venus human traps.

## Why It's Good For The Game
ThingAN1953

To clarify in more detail: Venus human traps spawn by flowering kudzu (with a 10% chance). However, they spawn immediately under the kudzu and can't be clicked on (even with the alt-click menu), so they can only be possessed by a ghost only if the kudzu above them gets destroyed or they get moved out of the kudzu.
Instead of just fiddling around with layers, moving venus human traps to the same ghost role system used by other ghost roles helps centralize them and make it more obvious that a) you can possess them to begin with, and b) how to possess them

## Changelog

:cl: RandomGamer123, bug originally reported by Murmic#2120 on Discord
qol: Makes it more clear that venus human traps can be possessed in the first place
fix: Makes it so that venus human traps can be possessed even if they are underneath kudzu
refactor: Moved venus human traps to the same spawning system that spiders used, now, a notification pops up when one is ready to be possessed, and you can possess it by using the ghost role menu, the link in chat, or by clicking on the notification
admin: Venus human traps now respect role bans (for ROLE_LAVALAND as its the default for ghost roles)
/:cl:

### CODE REVIEW

**Comment by BlueMemesauce** [comment]
Can you do the same thing for spider eggs too?

**Comment by RandomGamer123** [comment]
> Can you do the same thing for spider eggs too?

What do you mean by "the same thing"? Most of this code is based on spider egg code to begin with, so they already have most of the features here

**Comment by BlueMemesauce** [comment]
I thought spider eggs had a similar problem nevermind then

**Comment by RandomGamer123** [comment]
@tralezab I guess since you merged the previous kudzu-related PR

File: code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm [hunk 1]
```diff
@@ -55,21 +61,19 @@
 
 /obj/structure/alien/resin/flower_bud/Destroy()
 	QDEL_LIST(vines)
+	QDEL_NULL(countdown)
+	if(spawner) // anti harddel checks
+		spawner.flower_bud = null
+		if(!QDELETED(spawner))
+			qdel(spawner)
+		spawner = null
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm:69]: Why do we need to clear the flower_bud variable? shouldn't qdeling the spawner do that?

File: code/modules/mob_spawn/ghost_roles/venus_human_trap.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	var/obj/structure/alien/resin/flower_bud/flower_bud
+	/// Used to determine when to notify ghosts
+	var/ready = FALSE
+
+/obj/effect/mob_spawn/ghost_role/venus_human_trap/Destroy()
+	if(flower_bud) // anti harddel checks
+		flower_bud.spawner = null
+		if(!QDELETED(flower_bud))
+			qdel(flower_bud)
+		flower_bud = null
```
Comment (LemonInTheDark) [inline code/modules/mob_spawn/ghost_roles/venus_human_trap.dm:23]: similar question here

File: code/modules/mob_spawn/ghost_roles/venus_human_trap.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		return
+	if(href_list["activate"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost))
+			ghost.ManualFollow(src)
+			attack_ghost(ghost)
+
+/obj/effect/mob_spawn/ghost_role/venus_human_trap/allow_spawn(mob/user, silent = FALSE)
+	. = ..()
+	if(!.)
```
Comment (LemonInTheDark) [inline code/modules/mob_spawn/ghost_roles/venus_human_trap.dm:50]: Good protection against href exploits, +1

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
> Venus human traps now respect role bans (for ROLE_LAVALAND as its the default for ghost roles)

Is it default for ghost roles like spiders? or things that are actually on lavaland like golems

**Comment by RandomGamer123** [comment]
> > Venus human traps now respect role bans (for ROLE_LAVALAND as its the default for ghost roles)
> 
> Is it default for ghost roles like spiders? or things that are actually on lavaland like golems

```
///which role to check for a job ban (ROLE_LAVALAND is the default ghost role ban)
	var/role_ban = ROLE_LAVALAND
```
And looking at the code, ROLE_LAVALAND is also used for things like golems and fugitive hunters

> Why do we need to clear the flower_bud variable? shouldn't qdeling the spawner do that?

After testing this, yes this is not needed and removing it does not cause hard deletes.

**Comment by RandomGamer123** [comment]
Bump, @LemonInTheDark if you would want to take a look at this again now you're back

File: code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm [hunk 4]
```diff
@@ -55,21 +61,18 @@
 
 /obj/structure/alien/resin/flower_bud/Destroy()
 	QDEL_LIST(vines)
+	QDEL_NULL(countdown)
+	if(spawner) // anti harddel checks
+		if(!QDELETED(spawner))
+			qdel(spawner)
+		spawner = null
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm:68]: You think we can use QDEL_NULL here instead?

**Comment by RandomGamer123** [comment]
Bump to stop stale

**Pull request merged**